guides/specification.md

```diff
@@ -0,0 +1,549 @@
+
+That said, a fact declaration contains:
+- the fact name
+- the gatherer used to retrieve the fact
+- the argument(s) to be provided to the gatherer
+
+**Note:** 
+- many facts can be declared
+- all the declared facts would be registered in the [`facts`](#facts-1) namespaced evaluation scope.
+
+```yaml
+facts:
+  - name: <fact_name>
+    gatherer: <gatherer_name>
+    argument: <gatherer_argument>
+    
+  - name: <another_fact_name>
+    gatherer: <another_gatherer_name>
+    argument: <another_gatherer_argument>
+```
+
+The following example declares a **fact** named `corosync_token_timeout`, retrievable via the built-in `corosync.conf` **gatherer** to which will be provided the **argument** `totem.token` 
+```yaml
+facts:
+  - name: corosync_token_timeout
+    gatherer: corosync.conf
+    argument: totem.token
+
+  # other facts maybe
+```
+
+Finally, gathered facts, are used in Check's [Expectations](#expectations) to determine whether expected conditions are met for the best practice to be adhered.
+
+## Values
+
+Values are basically named variables that may evaluate differently based on the execution context and are used with Facts for _Contextual_ [Expectations](#expectations) Evaluation.
+
+> When contextual expectations is not needed, there's the following options available:
+> - use [**hardcoded**](#hardcoded-values) values
+> - define `values` as [**constants**](#constant-values)
+> 
+> Scenario:
+> 
+> No matter what the context is, the fact `awesome_fact` MUST always be `wanda`
+
+### Hardcoded Values
+
+Direct usage of a simple hadrcoded value
+```yaml
+expectations:
+  - name: awesome_expectation
+    expect: facts.awesome_fact == "wanda"
+```
+
+### Constant Values
+
+Define a Value with only the `default` specified (**omitting** `conditions`) for **constants** regardless of the context.
+```yaml
+values:
+  - name: awesome_constant_value
+    default: "wanda"
+
+expectations:
+  - name: awesome_expectation
+    expect: facts.awesome_fact == values.awesome_constant_value
+```
+
+### Contextual Values
+
+This is needed because the same check might expect facts to be treated differently based on the context.
+
+> Let's clarify with an example:
+>
+> A Check might define a fact named `awesome_fact` which is expected to be different given the _color_ of the execution.
+> - it has to be `cat` when the `color` in the execution context is `red`
+> - it has to be `dog` when the `color` in the execution context is `blue`
+> - it has to be `rabbit` in all other cases, regardless of the execution context
+> 
+> so we define a named variable `awesome_expectation` that resolves to `cat|dog|rabbit` when proper conditions are met
+>
+> allowing us to have an expectation like this 
+>
+> `expect: facts.awesome_fact == values.expected_token_timeout`
+
+In a nutshell, Values provide a way to decouple concepts, usually expeced values for facts, from their actual concretization during execution in certain scenarios.
```

Comment:
I find this sentense too complicated. Values are just variables.

Reply:
Gonna remove the sentence, it's redundant indeed.